datadog/resource_datadog_dashboard.go

```diff
@@ -2398,6 +2398,37 @@ func getLogStreamDefinitionSchema() map[string]*schema.Schema {
 			Optional: true,
 			Elem:     &schema.Schema{Type: schema.TypeString},
 		},
+		"show_date_column": {
+			Type:     schema.TypeBool,
+			Optional: true,
+		},
+		"show_message_column": {
+			Type:     schema.TypeBool,
+			Optional: true,
+		},
+		"message_display": {
+			Type:        schema.TypeString,
+			Optional:    true,
+			Description: "One of: ['inline', 'expanded-md', 'expanded-lg']",
+			ValidateFunc: func(val interface{}, key string) (warns []string, errs []error) {
+				value := val.(string)
+				switch value {
+				case "inline", "expanded-md", "expanded-lg":
+					break
+				default:
+					errs = append(errs, fmt.Errorf(
+						"%q contains an invalid value %q. Valid values are `inline`, `expanded-md`, or `expanded-lg`", key, value))
+				}
+				return
+			},
+		},
+		"sort": {
+			Type:     schema.TypeMap,
```

Comment:
Let's convert this to a `TypeList` since using the `Elem` block to define specific keys for the map is currently not possible. See https://www.terraform.io/docs/extend/schemas/schema-types.html#typemap. You can use a TypeList with a maxItem of 1 instead.